crates/biome_cli/src/commands/lint.rs

```diff
@@ -74,6 +86,14 @@ pub(crate) fn lint(
         &cli_options,
     )?;
 
+    if since.is_some() && !changed {
+        return Err(CliDiagnostic::incompatible_arguments("since", "changed"));
+    }
+
+    if changed {
+        store_changed_files(&mut session, &mut fs_configuration, since)?;
+    }
```

Comment:
OK, I think we are getting closer to what this feature will look like. I believe we should still apply some changes - apologies for that. It's a new feature with new functionalities that haven't yet, so we are essentially designing the feature as we go.

So, I think we don't need to modify the configuration. I think we should keep the paths emitted by the new function., and store them inside `TraversalMode`, very similar with what we do with `stdin`; then inside the `traverse.rs`, we create **new** `paths` that we construct from the `paths` passed via CLI. 

At the end, our traverse should process **only** the files that were emitted by git. In this way, we don't do a full scan of the FS, we only process the files we need. Does that make sense? Let me know if you have questions in case something isn't clear.

Reply:
Oh don't worry I know what I signed up for 😄 Sorry for the delay in response, I was preparing for some interviews, but now I have time to get back to this.

Based on what you said, I'll basically modify `store_changed_paths` to return the changed paths, call it from the `traverse` function and then run `traverse_inputs` only on those files.

Reply:
Exactly! 💯